internal/app/uploader.go

```diff
@@ -0,0 +1,85 @@
+package app
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"golang.org/x/crypto/sha3"
+)
+
+// BasinProviderUploader ...
+type BasinProviderUploader interface {
+	Upload(context.Context, string, string, io.Reader, *Signer, io.Writer) error
+	Reconnect() error
+}
+
+// BasinUploader contains logic of uploading Parquet files to Basin Provider.
+type BasinUploader struct {
+	namespace  string
+	relation   string
+	privateKey *ecdsa.PrivateKey
+	provider   BasinProviderUploader
+}
+
+// NewBasinUploader creates new uploader.
+func NewBasinUploader(ns string, rel string, bp BasinProviderUploader, pk *ecdsa.PrivateKey) *BasinUploader {
```

Comment:
new component that implements the logic of uploading a file. it has the `basinprovider` as a dependency